### PR TITLE
Remove types from reserved word list

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #2941

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -9629,8 +9555,6 @@ The following are reserved words:
 
     | `'async'`   <!-- ECMAScript2022 -->
 
-    | `'atomic_uint'`   <!-- GLSL -->
-
     | `'attribute'`   <!-- GLSL -->
 
     | `'auto'`   <!-- C++ -->
```
Comment (dj2) [inline wgsl/index.bs:9560]: Does `auto` count as a type given it's used where a type would be used?

File: wgsl/index.bs [hunk 2]
```diff
@@ -9649,8 +9571,6 @@ The following are reserved words:
 
     | `'cbuffer'`   <!-- HLSL -->
```
Comment (dj2) [inline wgsl/index.bs:9572]: Does this count as a buffer type?

File: wgsl/index.bs [hunk 3]
```diff
@@ -9709,26 +9629,16 @@ The following are reserved words:
```
Comment (dj2) [inline wgsl/index.bs:9629]: `dword` above is a type, yea?

File: wgsl/index.bs [hunk 4]
```diff
@@ -9959,50 +9761,6 @@ The following are reserved words:
```
Comment (dj2) [inline wgsl/index.bs:9761]: nil, null, nullptr

File: wgsl/index.bs [hunk 5]
```diff
@@ -10133,78 +9817,18 @@ The following are reserved words:
```
Comment (dj2) [inline wgsl/index.bs:9817]: Is `unorm` a type?

Comment (dneto0) [reply wgsl/index.bs:9817]: unorm is not a type.  When it's used it's as part of a context-dependent name for a channel in a texel format type.

e.g. rgba8unorm

File: wgsl/index.bs (hunk 4 above)
Comment (dneto0) [reply wgsl/index.bs:9759]: Sorry, are you asking for these to be reserved?  They already are.

(nil is one of only 5 keywords in Smalltalk)

File: wgsl/index.bs [hunk 6]
```diff
@@ -9527,92 +9527,18 @@ The following are reserved words:
...
 
     | `'GeometryShader'`   <!-- HLSL -->
 
     | `'Hullshader'`   <!-- HLSL -->
 
-    | `'InputPatch'`   <!-- HLSL -->
-
-    | `'LineStream'`   <!-- HLSL -->
-
-    | `'NULL'`   <!-- HLSL -->
```
[the remark this answers is not in the document]
Comment (dneto0) [reply wgsl/index.bs:9558]: Re-reserving NULL

File: wgsl/index.bs (hunk 2 above)
Comment (dneto0) [reply wgsl/index.bs]: unreserved now

File: wgsl/index.bs (hunk 3 above)
Comment (dneto0) [reply wgsl/index.bs:9627]: unreserved now.

**Review by alan-baker — APPROVED** [review]

File: wgsl/index.bs (hunk 4 above)
Comment (dj2) [reply wgsl/index.bs:9759]: I don't know what I was asking, heh.

File: wgsl/index.bs (hunk 5 above)
Comment (dj2) [reply wgsl/index.bs:9815]: Do we ever have it on its own?

Comment (dneto0) [reply wgsl/index.bs:9815]: No

**Pull request merged**